Modules/cjkcodecs/mappings_jp.h

```diff
@@ -591,7 +591,7 @@ __jisx0208_decmap+6950,33,38},{0,0,0},{0,0,0},{0,0,0},{0,0,0},{0,0,0},{0,0,0},
 };
 
 static const ucs2_t __jisx0212_decmap[6179] = {
-728,711,184,729,733,175,731,730,126,900,901,U,U,U,U,U,U,U,U,161,166,191,U,U,U,
+728,711,184,729,733,175,731,730,65374,900,901,U,U,U,U,U,U,U,U,161,166,191,U,U,U,
```

Comment:
https://github.com/python/cpython/blob/171ebb23426291834369fdf93813fd96b2bf88ff/Modules/cjkcodecs/mappings_jp.h#L1

Please take a look at the comment on the first line.
I 've not taken a look at the issue deeply, but if you want to change the mapping file, you should modify the generator.

Reply:
I totally missed that. I will do that.

Reply:
- Python generates the data using this as its source of truth: https://www.unicode.org/Public/MAPPINGS/OBSOLETE/EASTASIA/JIS/JIS0212.TXT.
- `encoding_rs`: also generated, but from https://encoding.spec.whatwg.org/index-jis0212.txt
- `iconv`: seems to be maintained indenpendently, and we can actually seen when they replaced TILDE with FULLWIDTH TILDE in their changelog: https://git.savannah.gnu.org/gitweb/?p=libiconv.git;a=blob;f=ChangeLog;h=0e2f5f00bc6aff84932cd92bd09a7c14f802c44d;hb=refs/heads/master#l4729

I couldn't figure out how Vim and Firefox proceed (although Vim does fallback to `iconv` for non-internally-supported encodings). I see two options:

1. add a hack in `genmap_japanese.py` around line 90 after the call to `loadmap(jisx0212file)` to reassign `jisx0212decmap[34][55] = ord('～')` (it does fix `__jisx0212_decmap` as expected and changes nothing else), and comment that properly
2. switch to WhatWG as the source of truth

1 can be done pretty quickly and with minimal risk of breaking anything. 2 has the advantage that it might help discover other issues in mappings, but is obviously much more involved.

If 1 seems reasonable to you, I will update this PR accordingly. If 2, I will close it and document this in the ticket for later.

Reply:
I have updated this MR with the hack in `genmap_japanese.py` and regenerated `mappings_jp.h`.

Reply:
I will take a look at which will be the better, If we decide to solve this issue, option 2 will be the better way.
But I need to check current status related to JIS0212.TXT